program.py

```diff
@@ -181,990 +160,207 @@ def send_omf_message_to_endpoint(message_type, message_omf_json, action='create'
+
+    # Send messages and check for each endpoint in config.json
+
     try:
-        print('------------------------------------------------------------------')
-        print(' .d88888b.  888b     d888 8888888888        8888888b. Y88b   d88P ')
-        print('d88P" "Y88b 8888b   d8888 888               888   Y88b Y88b d88P  ')
-        print('888     888 88888b.d88888 888               888    888  Y88o88P   ')
-        print('888     888 888Y88888P888 8888888           888   d88P   Y888P    ')
-        print('888     888 888 Y888P 888 888               8888888P"     888     ')
-        print('888     888 888  Y8P  888 888               888           888     ')
-        print('Y88b. .d88P 888   "   888 888               888           888     ')
-        print(' "Y88888P"  888       888 888      88888888 888           888     ')
-        print('------------------------------------------------------------------')
-
-        # Step 1
-        # OCS configuration
-        namespaceId = getConfig('Configurations', 'Namespace')
-        resourceBase = getConfig('Access', 'Resource')
-        tenant = getConfig('Access', 'Tenant')
-        apiversion = getConfig('Access', 'ApiVersion')
-        producerToken = getConfig('Credentials', 'ProducerToken')
-        clientId = getConfig('Credentials', 'ClientId')
-        clientSecret = getConfig('Credentials', 'ClientSecret')
-
-        # PI Web API configuration
-        dataServerName = getConfig('Configurations', 'DataServerName')
-        verify = getConfig('Configurations', 'VERIFY_SSL')
-        username = getConfig('Credentials', 'username')
-        password = getConfig('Credentials', 'password')
-
-        # shared configuration
-        resourceBase = getConfig('Access', 'Resource')
-
-        if verify is not None:
-            if verify == "False":
-                VERIFY_SSL = False
-
-        if not forceSending:
-            if tenant == "":
-                sendingToOCS = False
-            else:
-                sendingToOCS = True
-
-        if sendingToOCS:
-            checkBase = resourceBase + '/api/' + apiversion + \
-                '/tenants/' + tenant + '/namespaces/' + namespaceId
-            omfEndPoint = checkBase + '/omf'
-        else:
-            checkBase = resourceBase
-            omfEndPoint = checkBase + '/omf'
-
-        if not VERIFY_SSL:
-            print("You are not verifying the certificate of the end point.  This is not advised for any system as there are security issues with doing this.")
-
-        # Step 2
-        getToken()
-
-        # Steps 3-8 contained in here
-        oneTimeSendMessages()
-
-        # Step 9
+        # Send out the messages that only need to be sent once
+        for endpoint in endpoints:
+            if not endpoint["verify-ssl"]:
+                print("You are not verifying the certificate of the end point.  This is not advised for any system as there are security issues with doing this.")
+
+            get_token(endpoint)
```

Comment:
is this needed?

Reply:
@derekendres I don't think it is a bad idea to mention it, but the requests library does alert you if ssl is not verified so I suppose that it is redundant.